client/state/partner-portal/partner/reducer.ts

```diff
@@ -62,6 +63,15 @@ const current = ( state = initialState.current, action: AnyAction ) => {
 			}
 
 			return action.partner;
+
+		case JETPACK_PARTNER_PORTAL_PARTNER_SET_HAS_VALID_PAYMENT_METHOD:
+			if ( null !== state ) {
+				return Object.assign( state, {
+					has_valid_payment_method: action.hasValidPaymentMethod,
+				} );
+			}
+
+			return state;
```

Comment:
While this approach works, I think it might be better if instead of introducing a new action we use the already existing ones and add cases for them here e.g.:
```js
case STORED_CARDS_DELETE_COMPLETED:
    if ( deleted card was primary ) {
        return return Object.assign( state, {
            has_valid_payment_method: false,
        } );
    }

case STORED_CARDS_UPDATE_IS_PRIMARY_COMPLETED:
    return return Object.assign( state, {
        has_valid_payment_method: true,
    } );
```

This way we won't have to think about firing yet another action wherever payment methods are managed.

Reply:
Good note! I tried leveraging the already existing actions but I still had to add that action because when `STORE_CARDS_DELETE_COMPLETED` happen, it does not mean that the user does not have a valid payment method anymore, they might be other payment methods registered

Reply:
Update:
I could find a way to remove the need for creating a new action, so I updated the code with the refactor.